packages/labs/task/src/task.ts

```diff
@@ -351,3 +352,8 @@ export class Task<
       : args !== prev;
   }
 }
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+type MaybeReturnType<T extends undefined | ((...args: any) => any)> =
```

Comment:
More self contained, potentially easier on the compiler, and can avoid using `any` by writing:

```ts
type MaybeReturnType<F> = F extends (...args: unknown[])=>infer R ? R : undefined;
```

Reply:
thanks!